models/repo_permission.go

```diff
@@ -21,6 +21,8 @@ type Permission struct {
 	AccessMode perm_model.AccessMode
 	Units      []*repo_model.RepoUnit
 	UnitsMode  map[unit.Type]perm_model.AccessMode
+
+	User *user_model.User
```

Comment:
No, you should not depend a `User` here. It will make the dependencies cycle.

Reply:
Why? And should I add the user as an parameter/argument to the `CanWriteToBranch` func? Storing the user in the structs is an easy way to avoid a second parameter in the func.

Reply:
Yes, I think most functions should not be as the methods of the structs. `CanWriteToBranch` should be a method of `Permission`.

Reply:
Sorry for the late reply, but `CanWriteToBranch` is already a method of `Permission`. If I remove the user from there, should `CanWriteToBranch` take a `User` as an argument?

Reply:
That could be a standalone function.

Reply:
And which arguments it should take? A `User`, the branch name and a `Permission` object?

Reply:
I agree with lunny - should I refactor it?

Reply:
Since I don't know what you mean - feel free to do so.

Reply:
@lunny @qwerty287 done